framework/graphics/dx12_util.cpp

```diff
@@ -748,6 +748,32 @@ bool IsSoftwareAdapter(const format::DxgiAdapterDesc& adapter_desc)
     return software_desc;
 }
 
+bool VerifyAgilitySDKRuntime()
+{
+    char module_name[MAX_PATH] = {};
+    bool detected_runtime      = false;
+
+#if defined(D3D12_SUPPORT)
+    if (GetModuleFileNameA(nullptr, module_name, MAX_PATH))
```

Comment:
If this fails it may return MAX_PATH.
 https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-getmodulefilenamea  Could you check the return value to make sure it's less than MAX_PATH?